Fix eq3btsmart setting HVAC modes

bvweerd opened the pull request:
It seems like if a HVAC mode is the same as a preset, the HVAC mode change is ignored. in this PR the check which made the preset a higher priority than the HVAC mode has been removed, so a HVAC mode change can change the setting of the module to a preset as well.
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
Remove check if HVAC mode is the same as a preset, and therefore ignoring the HVAC mode change.

## Type of change
- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #66246
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

bvweerd committed:
Partly reverse preset incompatibility

It seems like some presets are unsupported by the native climate control of Home Assistant core. This change reverts the previous preset changes causing issues. It worked perfect with simple-thermostat custom lovelace card.

frenck commented:
> It seems like some presets are unsupported by the native climate control of Home Assistant core.

That doesn't make sense to me? Presets are FFA, as in, anything can be used and will do. Presets are not bound to any limitations.

frenck commented:
That sounds more like bug in `def set_preset_mode`, did you try pausing the debugger to see what is going on?

bvweerd commented:
So... it's going into the  _if self.preset_mode:_

Removing the if would solve the issue, but I have no clue why the if statement is there... any clue?

`    def set_hvac_mode(self, hvac_mode):
        """Set operation mode."""
        if self.preset_mode:
            return
        self._thermostat.mode = HA_TO_EQ_HVAC[hvac_mode]
`

bvweerd committed:
Remove priority of preset above HVAC mode 

If a preset was available of the given command, the hvac mode change was ignored. This can result in HVAC settings are ignored. By removing the check for a preset, the preset does not supersede the HVAC mode anymore

bvweerd committed:
Merge branch 'home-assistant:dev' into patch-1

bvweerd committed:
Revert "Partly reverse preset incompatibility"

This reverts commit 10fdc8eef457c369a042c631376ed33f29d0d8bf.

bvweerd committed:
Merge branch 'patch-1' of https://github.com/bvweerd/core into patch-1

bvweerd commented:
The previous commit removing the presets has been reverted. Instead the check which made the preset a higher priority than the HVAC mode has been removed, so a HVAC mode change can change the setting of the module to a preset as well.

rytilahti commented:
I found my test device and this indeed makes the buttons (at least seemingly) work again! I'm not sure why the valve is not being actuated even when the device reports new temperatures just fine (both on the device, and on responses to homeassistant) :-(

bvweerd commented:
>  I'm not sure why the valve is not being actuated [..] :-(

Sounds like a hardware issue (valves moving like a charm here...). Are you able to re-calibrate the valve, and is it moving then?

rytilahti commented:
> Sounds like a hardware issue (valves moving like a charm here...). Are you able to re-calibrate the valve, and is it moving then?

It could be a hardware issue but I think it was just that device thought it's already warm enough. Selecting the preset "open" will open the valve after some delay, and clicking the "off" button will close it again.

I'm fine for merging this, maybe the title could be updated to be more descriptive though?
